### ISSUE TITLE
php.lang.security.file-inclusion false positive

### ISSUE THREAD

**Issue by tuhaolam** [returntocorp/sgrep-rules#1262]
**Describe the bug**
php.lang.security.file-inclusion dont support Constant propagation

**To Reproduce**

- the code sample:
```php
$dir = "/tmp/";
include($dir . 'constant.php');
```
- the scan result
```
semgrep -f php/lang/security/file-inclusion.yaml  php/lang/security/file-inclusion.php 
include($dir . 'constant.php');

  1 rules:
ran 1 rules on 1 files: 1 findings
```

**Expected behavior**
should be ok without security alarm

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [x] P1: this will block me in the near future
- [] P2: annoying but not blocking me

**Additional Context**
no

**Comment by mschwager** [comment]
Hi there @tuhaolam,

I would expect our [constant propagation](https://semgrep.dev/docs/experiments/overview/#constant-propagation) to remove this FP. @aryx or @IagoAbal, does this functionality work for PHP?

**Comment by aryx** [comment]
Should work for PHP. Not sure why it does not trigger here. Maybe include is parsed in a special way and we don't visit
those constructs.

**Comment by IagoAbal** [comment]
I'm not familiar with the PHP parser but it seems to represent that small program as a string literal. This is a fragment of what I get with `semgrep-core -dump_ast`:
```OCaml
L(String(("$dir = "/tmp/";
include($dir . 'constant.php');
", ())
````

**Comment by mschwager** [comment]
> Maybe include is parsed in a special way and we don't visit those constructs.

I believe `include` is treated as a statement, not a function. This means the following is also valid PHP: 

```php
include $dir . 'constant.php';
```

Do I need to update the rule to consider this case, or should they be equivalent from the perspective of `semgrep-core`? If not, @IagoAbal or @aryx can I assign one of you to take a look?

**Comment by aryx** [comment]
In PHP you need to prefix your file with <?php
so the sample is really
```
<?php
$dir = "/tmp/";
include($dir . 'constant.php');
?>
```
if you use -dump_ast on this you'll get a different result.

@IagoAbal I'll assign it to you for now. Let me know if you have questions on PHP itself.

**Comment by mschwager** [comment]
Here's a snippet to repro: https://semgrep.dev/s/1nKw/

**Comment by mschwager** [comment]
It's fixed! https://semgrep.dev/s/1nKw/?version=develop

This fix will go out with the next release of Semgrep :+1:
